### PR TITLE
Menthy/form

### PR DESCRIPTION
**Pull request by menthy-wu**
Added default fields
added error toast on interal server error
fix button on interal server error

### CODE REVIEW

File: cypress/component/forms/Upload.cy.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import Upload from "@/components/dynamic/forms/Upload";
-import { USER } from "@/data/dynamic/forms/User";
-import { JUDGES } from "@/data/dynamic/forms/Judges";
+import { HELPER as USER } from "@/data/dynamic/forms/User";
```
Comment (shahdivyank) [inline cypress/component/forms/Upload.cy.jsx]: what's this for?

File: src/components/dynamic/forms/Committee.jsx [hunk 2]
```diff
@@ -2,23 +2,30 @@
...
-      toast(`✅ Submitted successfully!`);
-    });
+    axios
+      .post("/api/committees", committee)
+      .then(() => {
+        setLoading(false);
+        toast(`✅ Submitted successfully!`);
+      })
+      .catch(() => {
+        setLoading(false);
```
Comment (shahdivyank) [inline src/components/dynamic/forms/Committee.jsx]: do this in finally

File: src/components/dynamic/forms/Feedback.jsx [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 import { useState } from "react";
 import Form from "@/components/dynamic/forms/Form.jsx";
-import { FIELDS } from "../../../data/dynamic/forms/Feedback.js";
+import { FIELDS, HEALPER } from "../../../data/dynamic/forms/Feedback.js";
```
Comment (shahdivyank) [inline src/components/dynamic/forms/Feedback.jsx]: spelling

File: cypress/component/forms/Upload.cy.jsx [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
-import { USER } from "@/data/dynamic/forms/User";
-import { JUDGES } from "@/data/dynamic/forms/Judges";
 import { useState } from "react";
 
 describe("Forms Upload", () => {
   it("Upload valid PDF", () => {
     const file = "sample.pdf";
     const Parent = () => {
-      const [user, setUser] = useState(USER);
+      const [user, setUser] = useState({});
```
Comment (shahdivyank) [inline cypress/component/forms/Upload.cy.jsx]: why?

File: src/data/dynamic/forms/Feedback.js [hunk 5]
```diff
@@ -66,3 +67,13 @@ export const FIELDS = {
     width: 12,
   },
 };
+
+export const HELPER = {
```
Comment (shahdivyank) [inline src/data/dynamic/forms/Feedback.js]: call it questions

File: src/data/dynamic/forms/Judge.js [hunk 6]
```diff
@@ -95,3 +95,14 @@ export const FIELDS = {
...
+
+export const HELPER = {
+  name: "",
+  email: "",
+  phone: "",
+  gender: "",
+  shirt: "",
+  affiliation: "",
+  title: "",
+  photo: null,
```
Comment (shahdivyank) [inline src/data/dynamic/forms/Judge.js]: default should be empty string?! since its base64

File: cypress/component/Input.cy.jsx [hunk 7]
```diff
@@ -1,11 +1,11 @@
 import Input from "@/components/dynamic/Input";
 import { useState } from "react";
-import { USER } from "../../src/data/dynamic/forms/User";
+import { HELPER } from "../../src/data/dynamic/forms/Participant";
```
Comment (shahdivyank) [inline cypress/component/Input.cy.jsx]: 😭

File: src/app/api/volunteers/route.js [hunk 8]
```diff
@@ -13,21 +13,19 @@ import { authenticate } from "@/utils/auth";
 
 export async function POST(req) {
   const res = NextResponse;
-  const { auth } = await authenticate({
-    admins: 1,
-  });
+  const { auth, message, user } = await authenticate();
```
Comment (shahdivyank) [inline src/app/api/volunteers/route.js]: missing auth.post

File: src/components/dynamic/forms/Committee.jsx [hunk 9]
```diff
@@ -2,23 +2,25 @@
 
 import { useState } from "react";
 import Form from "@/components/dynamic/forms/Form.jsx";
-import { FIELDS } from "../../../data/dynamic/forms/Committees";
+import { FIELDS, ATTRIBUTES } from "../../../data/dynamic/forms/Committees";
```
Comment (shahdivyank) [inline src/components/dynamic/forms/Committee.jsx]: use @/src/data

File: src/app/api/participant/route.js [hunk 10]
```diff
@@ -5,9 +5,8 @@ import { authenticate } from "@/utils/auth";
 
 export async function POST(req) {
   const res = NextResponse;
-  // TODO: WHAT AUTH IS REQUIRED HERE
   const { auth, message, user } = await authenticate({
-    admins: 1,
+    participants: [-1, 0, 1],
```
Comment (shahdivyank) [inline src/app/api/participant/route.js]: this should be auth.post

File: src/app/api/team/route.js [hunk 11]
```diff
@@ -6,7 +6,7 @@ import { authenticate } from "@/utils/auth";
 export async function POST() {
   const res = NextResponse;
   const { auth, message, user } = await authenticate({
-    admins: 1,
+    participants: [-1, 0, 1],
```
Comment (shahdivyank) [inline src/app/api/team/route.js]: auth.post

File: src/app/api/team/route.js [hunk 12]
```diff
@@ -48,9 +48,7 @@ export async function POST() {
 
 export async function PUT(req) {
   const res = NextResponse;
-  const { auth, user } = await authenticate({
-    admins: 1,
-  });
+  const { auth, user } = await authenticate({ participants: [-1, 0, 1] });
```
Comment (shahdivyank) [inline src/app/api/team/route.js]: auth.put

File: src/app/api/team/route.js [hunk 13]
```diff
@@ -82,9 +80,7 @@ export async function PUT(req) {
 
 export async function GET(req) {
   const res = NextResponse;
-  const { auth } = await authenticate({
-    admins: 1,
-  });
+  const { auth } = await authenticate({ participants: [-1, 0, 1] });
```
Comment (shahdivyank) [inline src/app/api/team/route.js]: auth.get

File: cypress/component/Input.cy.jsx [hunk 14]
```diff
@@ -1,11 +1,10 @@
 import Input from "@/components/dynamic/Input";
 import { useState } from "react";
-import { USER } from "../../src/data/dynamic/forms/User";
 
 describe("Input", () => {
   it("Placeholder", () => {
     const Parent = () => {
-      const [user, setUser] = useState(USER);
+      const [user, setUser] = useState({});
```
Comment (shahdivyank) [inline cypress/component/Input.cy.jsx]: this should have at least the field ur trying to edit

File: src/components/dynamic/forms/Feedback.jsx [hunk 15]
```diff
@@ -2,10 +2,10 @@
 
 import { useState } from "react";
 import Form from "@/components/dynamic/forms/Form.jsx";
-import { FIELDS } from "../../../data/dynamic/forms/Feedback.js";
+import { FIELDS, ATTRIBUTES } from "../../../data/dynamic/forms/Feedback.js";
 
 const Feedback = () => {
-  const [feedback, setfeedback] = useState({});
+  const [feedback, setfeedback] = useState({ ATTRIBUTES });
```
Comment (shahdivyank) [inline src/components/dynamic/forms/Feedback.jsx]: should you spread here?

File: src/data/dynamic/user/Members.js [hunk 16]
```diff
@@ -0,0 +1,12 @@
+export const AUTH = {
+  POST: { participants: [1] },
+  GET: {
+    participants: [1],
```
Comment (shahdivyank) [inline src/data/dynamic/user/Members.js]: make it just 1, instead of [1]

File: src/data/dynamic/user/Team.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+export const AUTH = {
+  POST: { participants: [1] },
```
Comment (shahdivyank) [inline src/data/dynamic/user/Team.js]: make it just 1 instead of [1]

**Review by shahdivyank — APPROVED** [review]

**Review by SafeDuck — APPROVED** [review]

**Pull request merged**